app/src/main/java/com/odysee/app/ui/findcontent/FileViewFragment.java

```diff
@@ -3565,15 +3751,6 @@ private void checkOwnClaim() {
     }
 
     public static class StreamLoadErrorPolicy extends DefaultLoadErrorHandlingPolicy {
-        @Override
-        public long getRetryDelayMsFor(int dataType, long loadDurationMs, IOException exception, int errorCount) {
-            return exception instanceof ParserException
-                    || exception instanceof FileNotFoundException
-                    || exception instanceof Loader.UnexpectedLoaderException
-                    ? C.TIME_UNSET
-                    : Math.min((errorCount - 1) * 1000, 5000);
-        }
-
```

Comment:
This is removed because it is not compatible with Exoplayer 2.16.0, but to the best of my knowledge this was implementing the default behaviour for this method.

Reply:
This function is a custom error handler meant to keep trying to load the stream every 5 seconds if we didn't encounter one of the following fatal exceptions (resulting in an unrecoverable failure): `ParserException`, `FileNotFoundException` or `Loader.UnexpectedLoaderException`. If this is the default behaviour in Exoplayer 2.16.0, that's fine, otherwise we have to find an alternative way to implement this.

Reply:
```java
  /**
   * Retries for any exception that is not a subclass of {@link ParserException}, {@link
   * FileNotFoundException}, {@link CleartextNotPermittedException} or {@link
   * UnexpectedLoaderException}, and for which {@link
   * DataSourceException#isCausedByPositionOutOfRange} returns {@code false}. The retry delay is
   * calculated as {@code Math.min((errorCount - 1) * 1000, 5000)}.
   */
  @Override
  public long getRetryDelayMsFor(LoadErrorInfo loadErrorInfo) {
    IOException exception = loadErrorInfo.exception;
    return exception instanceof ParserException
            || exception instanceof FileNotFoundException
            || exception instanceof CleartextNotPermittedException
            || exception instanceof UnexpectedLoaderException
            || DataSourceException.isCausedByPositionOutOfRange(exception)
        ? C.TIME_UNSET
        : min((loadErrorInfo.errorCount - 1) * 1000, 5000);
  }
```

[This](https://github.com/google/ExoPlayer/blob/9f3868d6a07ba4baf1b73aa1239800f72ab9dacc/library/core/src/main/java/com/google/android/exoplayer2/upstream/DefaultLoadErrorHandlingPolicy.java#L101-L118) is the default implementation of this method, it considers `CleartextNotPermittedException` and `DataSourceException.isCausedByPositionOutOfRange(exception)` as non-retryable errors.

Do you think this is ok, or should the previous functionality be used?